src/cif.cc

```diff
@@ -268,3 +300,442 @@ freesasa_cif_structure_array(std::FILE *input,
+        if (i == 1)
+            block.set_pair(params_prefix + params_tags[i], gemmi::cif::quote(params_data[i]));
+        else
+            block.set_pair(params_prefix + params_tags[i], params_data[i]);
+    }
+}
+
+static std::string
+correct_inf_nan_values(const double value)
+{
+    if (std::isnan(value)) return ".";
+    if (std::isinf(value)) return "?";
+    return std::to_string(value);
+}
+
+static void
+append_freesasa_rsa_residue_to_block(gemmi::cif::Block &block, freesasa_node *residue)
+{
+    assert(freesasa_node_type(residue) == FREESASA_NODE_RESIDUE);
+
+    const freesasa_nodearea *abs, *reference;
+    freesasa_nodearea rel;
+
+    abs = freesasa_node_area(residue);
+    reference = freesasa_node_residue_reference(residue);
+
+    if (reference)
+        freesasa_residue_rel_nodearea(&rel, abs, reference);
+    else
+        rel = freesasa_nodearea_null;
+
+    std::string rsa_prefix{"_freeSASA_rsa."};
+    std::vector<std::string> rsa_tags{
+        "asym_id",
+        "seq_id",
+        "comp_id",
+        "abs_total",
+        "rel_total",
+        "abs_side_chain",
+        "rel_side_chain",
+        "abs_main_chain",
+        "rel_main_chain",
+        "abs_apolar",
+        "rel_apolar",
+        "abs_polar",
+        "rel_polar",
+    };
+
+    std::vector<std::string> rsa_data{
+        std::string{freesasa_node_name(freesasa_node_parent(residue))[0]},
+        std::string{freesasa_node_residue_number(residue)},
+        std::string{freesasa_node_name(residue)},
+        correct_inf_nan_values(abs->total),
+        correct_inf_nan_values(rel.total),
+        correct_inf_nan_values(abs->side_chain),
+        correct_inf_nan_values(rel.side_chain),
+        correct_inf_nan_values(abs->main_chain),
+        correct_inf_nan_values(rel.main_chain),
+        correct_inf_nan_values(abs->apolar),
+        correct_inf_nan_values(rel.apolar),
+        correct_inf_nan_values(abs->polar),
+        correct_inf_nan_values(rel.polar),
+    };
+
+    gemmi::cif::Loop *rsa_loop;
+    if (block.find(rsa_prefix, rsa_tags).ok()) {
+        rsa_loop = block.find(rsa_prefix, rsa_tags).get_loop();
+    } else {
+        gemmi::cif::Loop &temp_loop = block.init_loop(rsa_prefix, rsa_tags);
+        rsa_loop = &temp_loop;
+    }
+
+    rsa_loop->add_row(rsa_data);
+}
+
+static void
+append_freesasa_result_summary_to_block(gemmi::cif::Block &block, freesasa_node *result)
+{
+    assert(freesasa_node_type(result) == FREESASA_NODE_RESULT);
+
+    freesasa_node *structure = NULL, *chain = NULL;
+    const freesasa_nodearea *area = NULL;
+
+    std::string name{get_filename(freesasa_node_name(result))};
+
+    structure = freesasa_node_children(result);
+    assert(structure);
+
+    area = freesasa_node_area(structure);
+    assert(area);
+
+    std::string results_prefix{"_freeSASA_results."};
+    std::vector<std::string> result_tags{"source", "chains", "model", "atoms", "type", "surface_area"};
```

Comment:
I'm not sure, but the source should be clear from context here, so maybe not necessary?

Reply:
Yeah I was thinking the same. Im glad you agree.

Reply:
So if we remove the source, how do explicitly state the source when its UNKNOWN?

Reply:
I'm not sure I understand what the issue is. 🤔  What I meant is that the information about which PDB structure this is is encoded in other parts of the output, so need to state it explicitly as in the other output formats. What were you thinking of?

Reply:
So one of the options for source is UNKNOWN if I remove the source info. We won't know if the source is UNKNOWN. Are you find with that?

Reply:
Ok, so CIF has a source field somewhere that can be UNKNOWN?

Reply:
So I went back to find what I was referencing and I can't find it. Maybe I got it confused with something while writing the RSA block.  I agree with you its redundant. Ill remove it.

Reply:
Hmm, in the regular "log" output the source is output as unknown if there is no node_name for the result, at least. But I think in this context that's not necessary, as we have agreed 😄